### PR TITLE
Donovan barnyard testing

### PR DESCRIPTION
**Pull request by wmaxdonovan**
Merging the barnyard_testing branch into my long term donovan_SC_overhaul_remastered branch. This includes the Soil Water Fix, Output Standardization, the leap-year/mid-year start and end dates fix, and a number of bug fixes that have all been reviewed prior to this request.

### CODE REVIEW

File: Inputs/barnyard.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
                 "crop_type": "annual",
                 "fix_nitrogen": false,
 
-                "planting_date": 121,
+                "planting_date": 367,
```
Comment (mmotew) [inline Inputs/barnyard.json:49]: Just a thought: Maybe it's better to do this behind the scenes from the user? So, the user indicates in the json file that they do not want a crop to grow at all (e.g. sets a crop flag to 0), but the code simply knows to now set the planting date to 367 to get it to run correctly. Not sure this is the right thing to do or not. It's a question of if we want the hack to be in the json file or hidden from the json file. Ultimately the json file needs to be intuitive to a user. But, we may not be ready for that yet.

File: RUFAS/routines/soil/soil.py [hunk 2]
```diff
@@ -1,12 +1,154 @@
...
+            "Layer1":
+
+                "BottomDepth": 150,
+                "WiltingPoint": 0.1,
+                "FieldCapacity": 0.30,
+                "Saturation": 0.5,
+                "Ksat": 20,
+                "CationExclusionFraction": 0.0,
+                "Clay": 20,
+                "InitialTemperature": 15.77575,
```
Comment (mmotew) [inline RUFAS/routines/soil/soil.py:46]: Can you comment in what the units are for all these fields?

File: RUFAS/routines/soil/soil.py [hunk 3]
```diff
@@ -265,10 +406,10 @@ def __init__(self, layerName, layerData):
             self.bulkDensity = layerData['BulkDensity']
 
 
-            # Variables to calculate dailyEvapotranspiration
-            self.topEsoil = 0.0 # evaporation demand at top of layer
-            self.bottomEsoil = 0.0 # evaporation demand at bottom of layer
-            self.layerEsoil = 0.0 # evaporation demand at layer
+            # Variables to calculate daily Evapotranspiration
```
Comment (mmotew) [inline RUFAS/routines/soil/soil.py:409]: Why is Evapotranspiration capitalized?

File: RUFAS/routines/soil/soil_temp.py [hunk 4]
```diff
@@ -181,7 +181,7 @@ def calc_Tsurf(soil, crop, weather, time):
     Tbare = calc_Tbare(soil, crop, weather, time)
     bcv = calc_bcv(crop, time)
 
-    soil.Tsurf = (bcv * soil.Tsurf) + ((1 - bcv) * Tbare)
+    soil.Tsurf = (bcv * soil.listOfSoilLayers[0].temperature) + ((1 - bcv) * Tbare)
```
Comment (mmotew) [inline RUFAS/routines/soil/soil_temp.py:184]: Just to confirm:  is soil.Tsurf the same as the top layer soil temperature? Or, is Tsurf the air temperature at the ground surface, which is an input to the model.

File: RUFAS/simulation_engine.py [hunk 5]
```diff
@@ -93,6 +91,7 @@ def daily_simulation():
      #routines.daily_phosphorus_cycling_routine(state.soil, time, weather, config)
     routines.daily_animal_routine(state.animal, state.feed, weather, time)
     routines.daily_crop_routine(state.crop, weather, time, state.soil)
+    routines.daily_soil_routine(state.soil, state.crop, weather, time)
```
Comment (mmotew) [inline RUFAS/simulation_engine.py:94]: As long as all the variables used by the crop module are initialized properly, the order of calls here *shouldn't* matter. (But we'll see if that's the case.)

**Review by mmotew — COMMENTED** [review]
Looks good; left some individual comments. I think I've seen most of this already so I focused on the last major commit. If I'm wrong on that let me know.

File: Inputs/barnyard.json (hunk 1 above)
Comment (wmaxdonovan) [reply Inputs/barnyard.json:49]: I think you're right. I don't think we're quite there yet, but this is not a permanent solution. I'm mostly using it to short circuit the plant in the same way the spreadsheet model does.

File: RUFAS/routines/soil/soil.py (hunk 2 above)
Comment (wmaxdonovan) [reply RUFAS/routines/soil/soil.py:46]: That would be a good idea. These are taken directly from the JSon file, but I would be happy to track down units

File: RUFAS/routines/soil/soil.py (hunk 3 above)
Comment (wmaxdonovan) [reply RUFAS/routines/soil/soil.py:409]: good catch. It has been fixed.

File: RUFAS/routines/soil/soil_temp.py (hunk 4 above)
Comment (wmaxdonovan) [reply RUFAS/routines/soil/soil_temp.py:184]: Tsurf is initialized as the "Initial Temperature" of the first soil layer and then updated separately from first layer temperature as (bcv * first_layer_temp) + ((1 - bcv) * Tbare) where bcv is a bio-cover factor and Tbare is the temperature of bare soil under those conditions. The use of first layer temperature was likely confusion between surface temperature and the temperature of the first layer which the pseudocode treats differently. Taair is the representation of air temperature at the ground surface I believe.

**Pull request merged**